PatientWebApplication/HealthClinic.DAL/Service/PatientService.cs

```diff
@@ -84,13 +87,70 @@ public string ImageToSave(string path, FileModel file)
 
             }
         }
+
         /// <summary> This method is calling <c>PatientRepository</c> to find one patient using <paramref name="id"/>. </summary>
         /// <param name="id"><c>id</c> is <c>id</c> of a <c>PatientUser</c> that needs to be found.
         /// <returns>One patient</returns>
         public PatientUser GetOne(int id)
         {
             return PatientsRepository.FindOne(id);
         }
+
+        /// <summary> This method provides <c>PatientUser</c> <paramref name="patientId"/> and sends it to <c>PatientsRepository</c> there patient.IsBlocked will be set to true. </summary>
+        /// <param name="patientId"><c>PatientUser</c> is <c>PatientUser</c> that needs to be blocked.
+        /// </param>
+        /// <returns>null if PatientUser is not valid; otherwise, succesfully blocked PatientUser. </returns>
+        public PatientUser BlockPatient(int patientId)
+        {
+            PatientUser patient = PatientsRepository.FindOne(patientId);
+            return (patient == null) ? null : PatientsRepository.BlockPatient(patient);
+        }
+
+        /// <summary> This method is getting list of filtered malicious<c>PatientUser</c>.</summary>
+        /// <returns> List of filtered malicious patients. </returns>
+        public List<PatientUser> GetMaliciousPatients()
+        {
+            Dictionary<int,int> dict = new Dictionary<int, int>();
+            List<PatientUser> maliciousPatients = new List<PatientUser>();
+            List<DoctorAppointment> appointments = _regularAppointmentService.GetAll();
+            dict = GetCanceledAppointmentsInLastMonth(appointments);
+            return GetValidPatientsInLastMonth(dict);
```

Comment:
Write this in form:
"return GetValidPatientsInLastMonth(GetCanceledAppointmentsInLastMonth(_regularAppointmentService.GetAll()));"